modules/reindex/src/main/java/org/elasticsearch/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -167,7 +161,7 @@ public abstract class AbstractAsyncBulkByScrollAction<
         this.logger = logger;
         this.searchClient = searchClient;
         this.bulkClient = bulkClient;
-        this.threadPool = threadPool;
+        this.threadPool = searchClient.threadPool();
```

Comment:
Is there a difference between the `searchClient.threadPool()` and choosing `bulkClient.threadPool()`?

Reply:
No not really :) it's always the same instance.

Reply:
Would be nice to add an assertion that their thread pools are the same just to ensure

Reply:
I think that is `org.elasticsearch.threadpool.ThreadPool` that allows to access any actual `ExecutorService`.

Not sure I like this change. It hides the fact that we depend on the ThreadPool and instead "steals" it from the other dependency. Hypothetically, what happens if we change `searchClient` to not have the `threadPool` or at least not expose it any more?

Reply:
After seeing @idegtiarenko 's comment, I'm removing my approval, just to confirm the implications around thread pool is OK.

@original-brownbear , checking also other actions, I see that thread pool is indeed passed around a lot, even if there's a client argument. I am unaware of why there's both passed, if conceptually we could have taken the thread pool from the client. I just would like to ensure it's not a problem as well. I cannot think of any problems at the moment, only that perhaps the thread pools might be different (a different one is passed, different than the client's), but I'm not sure that's happening anywhere?